### ISSUE TITLE
Crash instead of error

### ISSUE THREAD

**Issue by joningold** [inkle/ink#291]
The following (broken) code crashes the run-time: it should generate a "attempting to use void value" error because append doesn't have a return code (or simply produce no choice):

```
VAR num = 0

*   {count(1)} BOOM 
-   -> END

=== function count(x) 
   ~ num += x
```

Note that changing to "~ num = x" makes the error go away.

**Comment by joningold** [comment]
Update: += appears to crash in general when used inside choices conditional functions:


```
*   {addToList()} Can't cope with the += in the function
-   -> END

VAR x = 0

=== function addToList()
    ~ x += 1 
    ~ return true
```

**Comment by joethephish** [comment]
Fixed in a45019520c2eb7851ed4de1d853659f2e9764250.
